Permanent/ViewControllers/ArchivesViewController.swift

```diff
@@ -219,67 +219,137 @@ class ArchivesViewController: BaseViewController<ArchivesViewModel> {
 
 extension ArchivesViewController: UITableViewDataSource, UITableViewDelegate {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return viewModel?.availableArchives.count ?? 0
+        var numberOfItemsInSection = 0
+        
+        if tableViewSections[section] == .pending {
+            numberOfItemsInSection = viewModel?.pendingArchives.count ?? 0
+        }
+        if tableViewSections[section] == .ok {
+            numberOfItemsInSection = viewModel?.selectableArchives.count ?? 0
+        }
+
+        return numberOfItemsInSection
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         var cell = UITableViewCell()
-        if let tableViewCell = tableView.dequeueReusableCell(withIdentifier: String(describing: ArchiveScreenDetailsTableViewCell.self)) as? ArchiveScreenDetailsTableViewCell,
-           let tableViewData = viewModel?.availableArchives {
-            let archiveVO = tableViewData[indexPath.row]
-            tableViewCell.updateCell(withArchiveVO: archiveVO, isDefault: archiveVO.archiveID == viewModel?.defaultArchiveId, isManaging: isManaging)
-            tableViewCell.rightButtonAction = { [weak self] cell in
-                var actions = [
-                    PRMNTAction(title: "Make Default".localized(), color: .primary, handler: { action in
-                        guard let archiveId = archiveVO.archiveID else { return }
-                        self?.showSpinner()
-                        self?.viewModel?.updateAccount(withDefaultArchiveId: archiveId, { accountVO, error in
-                            self?.hideSpinner()
-                            if error == nil {
-                                self?.updateCurrentArchive()
-                                tableView.reloadData()
-                            } else {
-                                self?.showAlert(title: .error, message: .errorMessage)
-                            }
+        if tableViewSections[indexPath.section] == .ok {
+            if let tableViewCell = tableView.dequeueReusableCell(withIdentifier: String(describing: ArchiveScreenChooseArchiveDetailsTableViewCell.self)) as? ArchiveScreenChooseArchiveDetailsTableViewCell,
+               let tableViewData = viewModel?.selectableArchives {
+                let archiveVO = tableViewData[indexPath.row]
+                tableViewCell.updateCell(withArchiveVO: archiveVO, isDefault: archiveVO.archiveID == viewModel?.defaultArchiveId, isManaging: isManaging)
+                tableViewCell.rightButtonAction = { [weak self] cell in
```

Comment:
Let's move the button actions to separate functions. This one is getting a little bit too cluttered

Reply:
check the button actions below as well

Reply:
Created `rightButtonAction() acceptButtonAction() declineButtonAction()` methods for tableview buttons.